RandBLAS/trig_skops.hh

```diff
@@ -0,0 +1,547 @@
+#include "RandBLAS/base.hh"
+#include "RandBLAS/exceptions.hh"
+#include "RandBLAS/random_gen.hh"
+#include "RandBLAS/sparse_skops.hh"
+#include "util.hh"
+
+#include <Random123/philox.h>
+#include <blas.hh>
+
+#include <iostream>
+#include <stdio.h>
+#include <stdexcept>
+#include <string>
+#include <tuple>
+
+#include <math.h>
+#include <typeinfo>
+#include <vector>
+
+#define MAX(a, b) (((a) < (b)) ? (b) : (a))
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+
+namespace RandBLAS::trig::kernels {
+// =============================================================================
+/// WARNING: None of the following functions or overloads thereof are part of the
+/// public API
+///
+
+// Generates a vector of Rademacher entries using the Random123 library
+template<SignedInteger sint_t = int64_t, typename state_t = RNGState<DefaultRNG>>
+state_t generate_rademacher_vector_r123(sint_t* buff, int64_t n, state_t &seed_state) {
+    DefaultRNG rng;
+    auto [ctr, key] = seed_state;
+
+    for (int64_t i = 0; i < n; ++i) {
+        typename DefaultRNG::ctr_type r = rng(ctr, key);
```

Comment:
This might raise a compiler error if ``state_t`` isn't ``RNGState<DefaultRNG>``.